src/confd/src/infix-services.c

```diff
@@ -307,6 +386,128 @@ static int web_change(sr_session_ctx_t *session, uint32_t sub_id, const char *mo
 	return put(cfg, srv);
 }
 
+/**
+ * Recursive directory remove (rm -r)
+ * @param file or directory to remove
+ *
+ * @returns 0 on success, else error
+ */
+
+int rm(const char *path) {
+    struct dirent *entry;
+    struct stat statbuf;
+    DIR *dir;
+    if (!path) {
+	    errno = EINVAL;
+	    return 1;
+    }
+
+    if (stat(path, &statbuf)) {
+	    return 1;
+    }
+
+    if (!S_ISDIR(statbuf.st_mode))
+	    return remove(path);
+
+    dir = opendir(path);
+    if (!dir) {
+	    return 1;
+    }
+
```

Comment:
Minor, nitpick -- select one style: either always use braces for all if/for/when with single statement, or always skip braces for single statements.